lib/database/migrations/20231011103554-create-lhc-period.js

```diff
@@ -0,0 +1,135 @@
+    const [newRunsData] = await queryInterface.sequelize.query(
+        'SELECT r.run_number, p.name as lhc_period \
+            FROM runs as r \
+            INNER JOIN lhc_periods as p \
+            ON p.id = r.lhc_period_id;',
+        { transaction },
+    );
+
+    return newRunsData.length ? `
+        INSERT INTO runs(run_number, lhc_period)
+        VALUES ${newRunsData.map(({ run_number, lhc_period }) => `(${run_number},'${lhc_period}')`).join(',')}
+        ON DUPLICATE KEY UPDATE
+            lhc_period = VALUES(lhc_period);
+        ` : '';
+};
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+        await queryInterface.createTable('lhc_periods', {
+            id: {
+                type: Sequelize.INTEGER,
+                allowNull: false,
+                primaryKey: true,
+                autoIncrement: true,
+            },
+            name: {
+                type: Sequelize.STRING,
+                unique: true,
+                allowNull: false,
+            },
+            created_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP'),
+            },
+            updated_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'),
+            },
+        }, { transaction });
+
+        await queryInterface.addColumn('runs', 'lhc_period_id', {
+            type: Sequelize.INTEGER,
+            reference: {
+                model: 'lhc_periods',
+                key: 'id',
+            },
+            allowNull: true,
+        }, { transaction });
+
+        const updateStmt = await prepareRunsUpdateStatement(
+            await createAndGetPeriods({ queryInterface, transaction }),
+            { queryInterface, transaction },
+        );
+        if (updateStmt) {
+            await queryInterface.sequelize.query(updateStmt, { transaction });
+        }
+
+        await queryInterface.removeColumn('runs', 'lhc_period', { transaction });
+    }),
+
+    down: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+        const { sequelize } = queryInterface;
+        await queryInterface.addColumn('runs', 'lhc_period', {
+            type: Sequelize.STRING,
+            allowNull: true,
+        }, { transaction });
+
+        const undoUpdateStmt = await prepareRunsUndoUpdateStatement({ queryInterface, transaction });
+        if (undoUpdateStmt) {
+            await sequelize.query(undoUpdateStmt, { transaction });
+        }
+
+        await queryInterface.removeColumn('runs', 'lhc_period_id', { transaction });
+        await queryInterface.dropTable('lhc_periods', { transaction });
+    }),
```

Comment:
Why do you do something so complex? You should try to keep things simple.
1- Create the table
2- `raw` Use INSERT SELECT to fill the table using run's periods value (https://mariadb.com/kb/en/insert-select/)
3- Create the column in runs table
4- `raw` UPDATE ... JOIN to set the `lhc_period_id` in runs table (https://stackoverflow.com/questions/47546152/how-to-update-a-row-based-a-joined-table-in-mariadb)
5- Delete the `lhc_period` column

No need for complex intermediate data passed in js (takes a lot of memory for not much)

Reply:
simplified

Reply:
See new comments